Microsoft.Alm.Authentication/TargetUri.cs

```diff
@@ -112,22 +107,71 @@ public bool IsBaseOf(TargetUri targetUri)
             if (targetUri == null)
                 return false;
 
-            return ResolvedUri.IsBaseOf(targetUri.ResolvedUri);
+            return ActualUri.IsBaseOf(targetUri.ActualUri);
         }
         /// <summary>
-        /// Gets a canonical string representation for the <see cref="ResolvedUri"/>.
+        /// Gets a canonical string representation for the <see cref="ActualUri"/>.
         /// </summary>
         /// <returns></returns>
         public override String ToString()
         {
-            return ResolvedUri.ToString();
+            return ActualUri.ToString();
+        }
+
+        public HttpClientHandler HttpClientHandler
+        {
+            get
+            {
+                bool useProxy = ProxyUri != null;
+                return new HttpClientHandler()
+                {
+                    Proxy = WebProxy,
+                    UseProxy = useProxy,
+                    MaxAutomaticRedirections = 2,
+                    UseDefaultCredentials = !useProxy
+                };
+            }
+        }
+
+        public WebProxy WebProxy
+        {
+            get
+            {
+                if (ProxyUri != null)
+                {
+                    WebProxy proxy = new WebProxy(ProxyUri);
+
+                    int dividerIndex = ProxyUri.UserInfo.IndexOf(':');
+
+                    if (!string.IsNullOrWhiteSpace(ProxyUri.UserInfo)
+                        && dividerIndex != -1)
+                    {
+                        string userName = ProxyUri.UserInfo.Substring(0, dividerIndex);
+                        string password = ProxyUri.UserInfo.Substring(dividerIndex + 1);
+
+                        NetworkCredential proxyCreds = new NetworkCredential(
+                            userName,
+                            password
+                        );
+
+                        proxy.UseDefaultCredentials = false;
+                        proxy.Credentials = proxyCreds;
+                    }
+
```

Comment:
Actually, in case 4 if you set UseDefaultCredentials to false after setting Credentials, these will be set to null.

This way will work:

```
httpClientHandler.Proxy.UseDefaultCredentials` = false;
httpClientHandler.Proxy.Credentials = new NetworkCredential(proxyUsername, proxyPassword);
```